### ISSUE TITLE
Why the martix can not pass the  `cusparseLtSpMMAPruneCheck`?

### ISSUE THREAD

**Issue by Mi-Peng** [NVIDIA/CUDALibrarySamples#120]
I try to prune the matrix into 2:4 sparsity myself as followed: 
```python
# test.py
    matrixA = torch.rand(8,8,dtype=torch.float32).cuda().contiguous()
    mask1 = torch.tensor([
        [1, 1, 0, 0, 1, 1, 0, 0],
        [1, 1, 0, 0, 1, 1, 0, 0],
        [1, 1, 0, 0, 1, 1, 0, 0],
        [1, 1, 0, 0, 1, 1, 0, 0],
        [1, 1, 0, 0, 1, 1, 0, 0],
        [1, 1, 0, 0, 1, 1, 0, 0],
        [1, 1, 0, 0, 1, 1, 0, 0],
        [1, 1, 0, 0, 1, 1, 0, 0],
    ]).cuda()
    mask2 = torch.tensor([
        [1, 0, 0, 1, 1, 0, 0, 1],
        [0, 1, 1, 0, 0, 1, 1, 0],
        [0, 1, 1, 0, 0, 1, 1, 0],
        [1, 0, 0, 1, 1, 0, 0, 1],
        [1, 0, 0, 1, 1, 0, 0, 1],
        [0, 1, 1, 0, 0, 1, 1, 0],
        [0, 1, 1, 0, 0, 1, 1, 0],
        [1, 0, 0, 1, 1, 0, 0, 1],
    ]).cuda()
    # calling the cusparseLt API `cusparseLtSpMMAPruneCheck` here
   checkprune(matrixA * mask1)
   checkprune(matrixA * mask2)
```
The difference between the two masks is that, there are two 0 in 4 consecutive elements in each **row** in mask1, while the **row** and the **column** in mask2 satisfy this constraint. Finally, **mask2 passes the check and mask1 does not**. 

So I'm confused about the constraint of 2:4 sparsity. Did I make a mistake in the constraint of 2:4 sparsity or was there a problem with my code. The data type of the tensor is `float32` and the initialization of cusparseLt are followed:

```cpp
// cusparseLt.cpp
auto order = CUSPARSE_ORDER_ROW;
auto opA   = CUSPARSE_OPERATION_NON_TRANSPOSE;
auto opB   = CUSPARSE_OPERATION_NON_TRANSPOSE;
auto type  = CUDA_R_32F;
unsigned alignment = 16;
auto compute_type  = CUSPARSE_COMPUTE_TF32;

... // init handle, matrix and so on 
CHECK_CUSPARSE( cusparseLtSpMMAPruneCheck(&handle, &matmul, dA_prunned, d_valid, stream) )
```

**Comment by fbusato** [comment]
`float` and `TF32`  support 50% sparsity, but the pattern is **1:2** not **2:4** as the other types. This is due to the internal organization of sparse tensor cores. Please see [cusparseLtPruneAlg_t](https://docs.nvidia.com/cuda/cusparselt/types.html#cusparseltprunealg-t)
